### PR TITLE
backend/vs: Add include directories to projects as well

### PR DESCRIPTION
**Pull request by vvainola**
- "Go To Document" action previously only worked on c/cpp files which had the include directories set but it was not possible to move from header to another header.

Fixes same thing as in https://github.com/mesonbuild/meson/pull/7385

### CODE REVIEW

File: mesonbuild/backend/vs2010backend.py [hunk 1]
```diff
@@ -966,9 +966,7 @@ def gen_vcxproj(self, target, ofname, guid):
             t_inc_dirs = [self.relpath(self.get_target_private_dir(target),
                                        self.get_target_dir(target))]
             if target.implicit_include_directories:
-                t_inc_dirs += ['.']
-            if target.implicit_include_directories:
-                t_inc_dirs += [proj_to_src_dir]
+                t_inc_dirs += ['.', proj_to_src_dir]
```
Comment (nirbheek) [inline mesonbuild/backend/vs2010backend.py:969]: This change seems unrelated, should be split into a new commit.

File: mesonbuild/backend/vs2010backend.py [hunk 2]
```diff
@@ -983,9 +981,8 @@ def gen_vcxproj(self, target, ofname, guid):
                     else:
                         define = arg[2:]
                     # De-dup
-                    if define in file_defines[l]:
-                        file_defines[l].remove(define)
-                    file_defines[l].append(define)
+                    if define not in file_defines[l]:
+                        file_defines[l].append(define)
```
Comment (nirbheek) [inline mesonbuild/backend/vs2010backend.py:985]: This change also seems unrelated, but it's cleanup like the above change, so can be clubbed with it.

File: mesonbuild/backend/vs2010backend.py [hunk 3]
```diff
@@ -1025,8 +1025,6 @@ def gen_vcxproj(self, target, ofname, guid):
         if len(target_args) > 0:
             target_args.append('%(AdditionalOptions)')
             ET.SubElement(clconf, "AdditionalOptions").text = ' '.join(target_args)
-
-        target_inc_dirs.append('%(AdditionalIncludeDirectories)')
```
Comment (nirbheek) [inline mesonbuild/backend/vs2010backend.py:1029]: I am not sure I understand why this has been removed.

**Review by nirbheek — COMMENTED** [review]
Also, could you please make sure that your commit message body is wrapped at 70 characters? That's the standard with git. Thanks!

File: mesonbuild/backend/vs2010backend.py (hunk 1 above)
Comment (vvainola) [reply mesonbuild/backend/vs2010backend.py:969]: Split cleanup into own commit.

File: mesonbuild/backend/vs2010backend.py (hunk 3 above)
Comment (vvainola) [reply mesonbuild/backend/vs2010backend.py:1029]: Split cleanup into own commit.

File: mesonbuild/backend/vs2010backend.py (hunk 2 above)
Comment (vvainola) [reply mesonbuild/backend/vs2010backend.py:985]: Split cleanup into own commit.

**Comment by nirbheek** [comment]
Thanks!

**Comment by xclaesse** [comment]
This seems to have broken CI in master.

**Comment by nirbheek** [comment]
This code is not called on MSYS2 at all.
